Lib/tkinter/__init__.py

```diff
@@ -3745,27 +3745,34 @@ def compare(self, index1, op, index2):
         return self.tk.getboolean(self.tk.call(
             self._w, 'compare', index1, op, index2))
 
-    def count(self, index1, index2, *options): # new in Tk 8.5
+    def count(self, index1, index2, *options, return_ints=False): # new in Tk 8.5
         """Counts the number of relevant things between the two indices.
 
         If INDEX1 is after INDEX2, the result will be a negative number
         (and this holds for each of the possible options).
 
         The actual items which are counted depends on the options given.
         The result is a tuple of integers, one for the result of each
-        counting option given, if more than one option is specified,
-        otherwise it is an integer. Valid counting options are "chars",
-        "displaychars", "displayindices", "displaylines", "indices",
-        "lines", "xpixels" and "ypixels". The default value, if no
-        option is specified, is "indices". There is an additional possible
-        option "update", which if given then all subsequent options ensure
-        that any possible out of date information is recalculated."""
+        counting option given, if more than one option is specified or
+        return_ints is false (default), otherwise it is an integer.
+        Valid counting options are "chars", "displaychars",
+        "displayindices", "displaylines", "indices", "lines", "xpixels"
+        and "ypixels". The default value, if no option is specified, is
+        "indices". There is an additional possible option "update",
+        which if given then all subsequent options ensure that any
+        possible out of date information is recalculated.
+        """
         options = ['-%s' % arg for arg in options]
         res = self.tk.call(self._w, 'count', *options, index1, index2)
         if not isinstance(res, int):
```

Comment:
I am trying to understand the total code here.  The check for int implies that when want_objects is true, a single object is returned as an int.  Are multiple objects returned as a tuple or list?  If the former, it would seem that the check should be for 'isinstance(res, str)'.

Reply:
If wantobjects is 1, the single count is returned as an int, and multiple counts are returned as a tuple of ints.
If wantobjects is 0, the result is a string.

So yes, this condition can currently be written as `isinstance(res, str)`. There may be arguments for and against both variants. I am planning to introduce an experimental mode in which the result will always be a Tcl_Obj, forcing the higher level code to do explicit conversion, then `not isinstance(res, int)` will continue to work.